spec/integration/client/client_spec.rb

```diff
@@ -485,7 +485,7 @@ class ::Blah
     it "should have a cookbook attribute" do
       result = shell_out("#{chef_client} -c \"#{path_to("config/client.rb")}\" -o 'x::default' --no-fork", cwd: chef_dir)
       result.error!
-      expect(result.stdout).to include('COOKBOOKS: {"x" => {"version" => "0.0.1"}}')
+      expect(result.stdout).to include('COOKBOOKS: {"x"=>{"version"=>"0.0.1"}}')
```

Comment:
@johnmccrae  this will make all other tests fail because ruby 3.4 has spaces in the output. this was fixed on the base ruby upgrade branch but since the linux plan is still on ruby 3.1 in this branch, linux plan is failing.

Reply:
We can update the spec instead so that Hashes are normalized, that will make specs happy for 3.1+ (considering habitat package still has ruby3.1), since this is literal string match in the spec.
Something like 
```suggestion
      normalized_stdout = result.stdout.gsub(/(\w+)\s*=>\s*/, '\1=>')
      expect(normalized_stdout).to include('COOKBOOKS: {"x"=>{"version"=>"0.0.1"}}')
```

Reply:
yeah that or just do a regex match if you don't want to modify the original output for the check

```suggestion
      expect(result.stdout).to match(/COOKBOOKS: \{"x"\s*=>\s*\{"version"\s*=>\s*"0\.0\.1"\}\}/)
```